packages/data-broker/src/services/dhis/getDhisApiInstance.js

```diff
@@ -5,13 +5,30 @@
 
 import { DhisApi } from '@tupaia/dhis-api';
 import { getDhisConfig } from '@tupaia/utils';
+import { DhisInputSchemeResolvingApi } from './DhisInputSchemeResolvingApi';
 
 const instances = {};
 
-export const getDhisApiInstance = options => {
+export const getDhisApiInstance = (options, models) => {
+  console.log('getDhisApiInstance');
   const { serverName, serverUrl } = getDhisConfig(options);
   if (!instances[serverName]) {
-    instances[serverName] = new DhisApi(serverName, serverUrl);
+    const api = new DhisApi(serverName, serverUrl);
+
+    const dhisInputSchemeResolvingApi = new DhisInputSchemeResolvingApi(models, api);
+
+    const proxy = new Proxy(api, {
```

Comment:
Ran into some sort of circular import problem, not sure how... but the proxy pattern seems to work well, can debug as usual etc. Might make code navigation harder through IDEs but when this code is cleaned up it should be simple enough for a human to follow.

Reply:
Wow, I didn't know this was baked in to js - that's super cool 😎

Reply:
What happens if you just pass in `dhisInputSchemeResolvingApi` as the handler directly? From a brief read of the mdn doc for Proxy I would've thought that would work

Reply:
No, it doesn't work, it needs the proxy specification object, with get() etc.